pkg/virt-handler/virtwrap/host.go

```diff
@@ -0,0 +1,68 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2017 Red Hat, Inc.
+ *
+ */
+
+package virtwrap
+
+import (
+	"time"
+	// Currenlty only libvirt is supported and there is not desire
+	// so-far to support anything else, but let's have a clear
+	// design.
+	"kubevirt.io/kubevirt/pkg/virt-handler/virtwrap/libvirt"
+)
+// Defining the host interface.
+//
+// Each hypervisor drivers will have to implmement this interface. It
+// will represents the only public solution for the other components
+// to comunicate with internal hypervisor functionalities
+
+type Host interface {
```

Comment:
Would it make sense to call that all `Hypervisor` instead of `Host`?

Reply:
Yes probably, lets use `Hypervisor` if everyone are agreed.

Reply:
Well thinking a bit more I do like `Host` because it's a concise name, if you don't have strong discord I would prefer keep it. Sounds good?

Reply:
I don't have a strong opinion here. What I would just keep in mind is, that in the k8s eco-system we have a `Node` which represents a physical or virtual `Host`. We would then also have a `Host` which is our hypervisor driver, which is only as much related to the host, as it runs on the host where virt-handler runs. There are also additions kubevirt native features, regarding to e.g. storage, which are also host local, but not necessarily part of our Host interface (e.g. cloud-init volumes, docker registry images, ...).

In the context of that, I am just not sure if `Host` would really be the wises choice, but I can live with it.

Reply:
Fair enough You made that clear `Hypervisor` is what we want.